### PR TITLE
chore: implement identity skeleton structure

### PR DESCRIPTION
**Pull request by Ben-Sheppard**
## Description

This PR introduces a temporary structure and space for Identity within the monorepo to allow the Identity team to begin implementation of the required minimal functionality. It is likely (almost certain) that as the project evolves this structure will also change.

### CODE REVIEW

File: identity/frontend/package.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
+{
```
Comment (dlavrenuek) [inline identity/frontend/package.json:1]: ❌ Let's remove the `frontend` folder for now. It only contains (here) unused commands/dependencies. I would create the frontend structure at a later stage

File: identity/backend/Makefile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+license-check:
+	mvn license:check
+
+license-fix:
+	mvn license:format
+
+test: license-check
+	mvn clean verify
+
+generate-dependency-docs:
```
Comment (dlavrenuek) [inline identity/backend/Makefile:22]: ❌ I'm not sure that we will have dedicated dependency list for Identity and would expect it being part of the monojar build. So I would remove this script as well as the template

File: identity/backend/pom.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-autoconfigure</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-context</artifactId>
```
Comment (dlavrenuek) [inline identity/backend/pom.xml:37]: ❓ This dependency does not seem to be used in the app. Let's add what we need as we go

**Review by dlavrenuek — CHANGES_REQUESTED** [review]
Thank you for kicking this off 🚀 I have added some suggestions

File: identity/backend/src/main/java/io/camunda/identity/Application.java
Comment (maryarm) [inline identity/backend/src/main/java/io/camunda/identity/Application.java:1]: ❓ I'm not sure if we need a Spring boot application for identity or should it be part of the higher directory to be used with all modules?

File: identity/backend/pom.xml (hunk 3 above)
Comment (Ben-Sheppard) [reply identity/backend/pom.xml:37]: I originally didn't have this here but dependency analyser told me that it needed to be in place, I'll double check though

File: identity/backend/src/main/java/io/camunda/identity/Application.java
Comment (dlavrenuek) [reply identity/backend/src/main/java/io/camunda/identity/Application.java:1]: You are absolutely correct, we don't need a spring boot app. For now I would call it a temporal step, so we can start hacking (and start Identity) without the need to build and start Zeebe locally. This should not lead to problems in the global scope tho, for example if there are two spring boot apps defined.

Comment (Ben-Sheppard) [reply identity/backend/src/main/java/io/camunda/identity/Application.java:1]: Initially I added the spring boot application part as a misunderstanding however I've clarified with @dlavrenuek that its something we should have to support our development efforts (starting our app vs integration with Zeebe and starting all of Zeebe for example), in the long term this application file will be removed.

File: identity/backend/pom.xml (hunk 3 above)
Comment (Ben-Sheppard) [reply identity/backend/pom.xml:37]: Yea confirmed locally that the dependency analyzer thinks this dependency is required here so I've left it in for now

**Review by maryarm — APPROVED** [review]
LGTM :)

**Review by dlavrenuek — APPROVED** [review]
Looks good 👍

**Pull request merged**